Make slider handles stick to the cursor

adroitwhiz opened the pull request:
### Resolves

Resolves #1202

### Proposed Changes

This PR adjusts the color slider position calculation to take into account where the user first clicked on the handle.

### Reason for Changes

This prevents sliders from "jumping" when the user starts dragging them:

Before | After
--- | ---
![Peek 2020-07-31 12-26](https://user-images.githubusercontent.com/25993062/89056328-d93d1400-d329-11ea-8d0d-df73fbbf15ef.gif) | ![Peek 2020-07-31 12-32](https://user-images.githubusercontent.com/25993062/89056383-ee19a780-d329-11ea-910d-b384416f9493.gif)


### Test Coverage

Tested manually.

adroitwhiz committed:
Make slider handles stick to the cursor

fsih commented:
I'm seeing an issue where the handle jumps a bit when I mouse up
![jump](https://user-images.githubusercontent.com/2855464/92158665-41d05280-edfa-11ea-890b-348d6a8b56be.gif)

adroitwhiz committed:
Don't treat slider handles as background clicks

adroitwhiz committed:
Make slider gradients match actual slider range

adroitwhiz commented:
The slider jumping on mouse up has been fixed.

There's another issue that I introduced and have fixed as well, and it affects the way sliders are displayed. Essentially, because we're treating the center of the slider handle as the value, the slider's rounded end caps are outside its effective "range":
![image](https://user-images.githubusercontent.com/25993062/94752908-877d2e00-035a-11eb-8597-5d8e6abcd58f.png)

When the slider is all the way to the left, for instance, the center of the slider handle is still 13 pixels to the right of where the slider starts.

Because of this, I changed the way the slider gradients are displayed so that the rounded end caps are solid, and the gradient only transitions in the region shown above.

fsih reviewed (approved):
Looks great! Thanks!
